benchmarks/graphics_pipeline/main.cpp

```diff
@@ -455,8 +502,24 @@ void ProjApp::ProcessInput()
     }
 }
 
+// Generates `m` random indices without repetition from [0, 1, ..., `n`-2, `n`-1]
+std::vector<uint32_t> GenerateRandomIndices(uint32_t m, uint32_t n)
+{
+    PPX_ASSERT_MSG(m <= n, "Invalid values. `m` must be less or equal to `n`.");
+    std::vector<uint32_t> indices(n);
+    std::iota(indices.begin(), indices.end(), 0);
+    std::shuffle(indices.begin(), indices.end(), std::random_device{});
+    return {indices.begin(), indices.begin() + m};
+}
+
 void ProjApp::Render()
 {
+    mCurrentSphereCount = pSphereInstanceCount->GetValue();
+    if (mCurrentSphereCount != mPreviousSphereCount) {
+        mSphereIndices       = GenerateRandomIndices(mCurrentSphereCount, kMaxSphereInstanceCount);
```

Comment:
i think we would need the spheres to be deterministic so that when we can compare the profiling result across different runs.

Reply:
Done.